pallets/creditcoin/src/ocw.rs

```diff
@@ -197,7 +199,13 @@ impl<T: Config> Pallet<T> {
 		tx_id: &ExternalTxId,
 	) -> VerificationResult<T::Moment> {
 		let rpc_url = blockchain.rpc_url()?;
-		let tx = rpc::eth_get_transaction(tx_id, &rpc_url)?;
+		let tx = rpc::eth_get_transaction(tx_id, &rpc_url).map_err(|e| {
+			if let RpcError::NoResult = e {
```

Comment:
this is small, but wouldn't this be more readable as `e = RpcError::NoResult`?

Reply:
It wouldn't be equivalent, because the left-hand side is a pattern and the right hand is an expression, so

```rust
if let e = RpcError::NoResult { /*stuff*/ } else { /*stuff*/ }
```

 would be equivalent to 

```rust
match e {
    e => {  /*stuff*/ }
    // this arm is unreachable, because the first arm matches anything and binds it to the variable `e`
    _ => { /*stuff*/}
}
```

(I also may have misunderstood what you meant)